index.js

```diff
@@ -71,38 +71,52 @@ var FloatLabelTextField = React.createClass({
     };
   },
 
-  withBorder: function() {
-    if (!this.props.noBorder) {
-      return styles.withBorder;
-    };
+  componentWillReceiveProps: function(newProps) {
+    console.log('componentWillReceiveProps');
+    this.setState({
+      text: newProps.value
+    });
+  },
+
+  getAdditionalStyle: function(name) {
+    if (this.props.style && this.props.style[name])
+    {
+      return this.props.style[name];
+    }
+
+    return {};
```

Comment:
I don't think returning a bool will help here .. What is the behavior of adding a null or undefined style going to cause?